lib/workers/repository/finalise/repository-statistics.ts

```diff
@@ -0,0 +1,60 @@
+import type { RenovateConfig } from '../../../config/types';
+import { logger } from '../../../logger';
+import { getPrCache } from '../../../modules/platform/github/pr';
+import { PrState } from '../../../types';
+import { GithubHttp } from '../../../util/http/github';
+
+const githubApi = new GithubHttp();
+const renovateBot = 'renovate[bot]';
+
+type PrStats = {
+  total: number;
+  open: number;
+  closed: number;
+  merged: number;
+};
+
+function initStats(): PrStats {
+  return {
+    total: 0,
+    open: 0,
+    closed: 0,
+    merged: 0,
+  };
+}
+
+export async function runRenovateRepoStats(
+  config: RenovateConfig
+): Promise<void> {
+  const prStats = initStats();
+  const prCache = await getPrCache(
```

Comment:
I'm not sure if it's okay to rely on it.
- It's platform-dependent and GitHub-only for now
- Its structure is subject to change
- Users may want to see *delta* values, i.e. how many PRs are open/closed/merged for this particular run. Seems to be complicated to implement with repository cache.

Most likely, it won't be much harder to init/update stats counters manually after relevant platform actions. Something similar is already done for HTTP statistics.

Reply:
it is intended for github use only, ive added the required checks before calling the newly added function. thanks for the heads up.

you might be right, will wait for some further instruction.

other than that, ive added some more changes to files related to github which are needed for this to work. can u please check out https://github.com/renovatebot/renovate/pull/15319

Reply:
as @MaronHatoum found and pointed out, on the first run of renovate the pr are filtered and storted by config.renovateusername then stored with repo as key.

on the second run we are getting 304 code and the previous cahced list is returned.
therefore we need to add support for pr created by [bot] and retrieve them later on